NTupleAnalyzerHH_nanoAOD.py

```diff
@@ -1094,132 +1158,281 @@ def LeptonsAndJetsFromHH(T):
+					dr = abs(m.DeltaR(g))
+					if dr<mindr:
+						mindr =dr
+						bestrecojetind=ind
+				if mindr<0.4:
+					matchedrecojetsH.append(jetsH[bestrecojetind])
+					recojetindsH.append(recojetOriIndsH[bestrecojetind])
+					usedHjetsInd.append(bestrecojetind)
+				else:
+					matchedrecojetsH.append(emptyvector)
+					recojetindsH.append(-99)
+			while len(recojetindsH)<2 :
+				recojetindsH.append(-99)
+
+###################### AK4 matching with 2 GEN GETS from Z ##########################
+			usedZjetsInd = []
+			for g in genjetsZ:
+				bestrecojetind=-1
+				mindr = 99999
+				ind=-1
+				for m in jetsZ:
+					ind+=1
+					skip = False
+					for indc in range(len(usedZjetsInd)):
+						if ind == usedZjetsInd[indc]:
+							skip = True
+							continue
+					for indc in range(len(usedHjetsInd)):
+						if ind == usedHjetsInd[indc]:
+							skip = True
+							continue
+					if skip: continue
+					dr = abs(m.DeltaR(g))
+					if dr<mindr:
+						mindr =dr
+						bestrecojetind=ind
+				if mindr<0.4:
+					matchedrecojetsZ.append(jetsZ[bestrecojetind])
+					recojetindsZ.append(recojetOriIndsZ[bestrecojetind])
+					usedZjetsInd.append(bestrecojetind)
+				#recojetindsZ.append(bestrecojetind)
+				else:
+					matchedrecojetsZ.append(emptyvector)
+					recojetindsZ.append(-99)
+			while len(recojetindsZ)<2 :
+				recojetindsZ.append(-99)
+
+###################### AK8 JETS #####################################################
+		if radius == 8:
+###################### AK8 matching with 2 GEN GETS from HIGGS ######################
+			usedHjetsInd = []
+			mindr1 = 99999
+			mindr2 = 99999
 			bestrecojetind=-1
-			mindr = 99999
-			ind=-1
-			for m in jetsH:
-				ind+=1
-				if abs(T.Jet_partonFlavour[recojetOriIndsH[ind]]) != 5 : continue
-				#if v == '': print 'ind in jetsH', ind ,'reco idex', recojetOriIndsH[ind], 'flavor', T.PFJetPartonFlavourAK4CHS[recojetOriIndsH[ind]]
+			bestindices = [-1,-1]
+
+			for j in range(len(jetsH)):
+
 				skip = False
 				for indc in range(len(usedHjetsInd)):
-					if ind == usedHjetsInd[indc]:
+					if j == usedHjetsInd[indc]:
 						skip = True
-						#if v == '':print ' >> skip ind in jetsH', ind
 						continue
 				if skip: continue
-				dr = abs(m.DeltaR(g))
-				#if v == '':print 'ind in jetsH', ind ,'DeltaR:',dr, ' >> gen pt eta', g.Pt(),g.Eta(),' >> reco pt eta', m.Pt(),m.Eta(), 'dr', dr
-				if dr<mindr:
-					mindr =dr
-					bestrecojetind=ind
-			if mindr<0.5: # was 0.6
-				matchedrecojetsH.append(jetsH[bestrecojetind])
+
+				subjet1,subjet2 = [],[]
+
+				if SubJetInd1[j] == -1:
+					sub1 = emptyvector
+				else:
+					sub1 = TLorentzVector()
+					sub1.SetPtEtaPhiM(SubJetPt[SubJetInd1[j]],SubJetEta[SubJetInd1[j]],SubJetPhi[SubJetInd1[j]],SubJetMass[SubJetInd1[j]])
+
+				if SubJetInd2[j] == -1:
+					sub2 = emptyvector
+				else:
+					sub2 = TLorentzVector()
+					sub2.SetPtEtaPhiM(SubJetPt[SubJetInd2[j]],SubJetEta[SubJetInd2[j]],SubJetPhi[SubJetInd2[j]],SubJetMass[SubJetInd2[j]])
```

Comment:
This looks like a problem in the indentation compared to the old version, please check.